### PR TITLE
Implemented refresh storage data in DB, If a storage resource is missing in ODIMRA and present in device

### PR DESCRIPTION
**Pull request by amar-shalgar**
Implemented refresh storage data in DB, If a storage resource is missing in ODIMRA and present in device
Modified code in svc-aggregation and svc-systems

Issue No. #129

### CODE REVIEW

**Review by skhan-anjum — APPROVED** [review]

**Review by a-ajith — APPROVED** [review]

File: svc-aggregation/system/common.go [hunk 1]
```diff
@@ -668,25 +759,28 @@ func createServerSearchIndex(computeSystem map[string]interface{}, oidKey, devic
 	}
 
 	// saving the firmware version
-	if firmwareVersion := getFirmwareVersiomn(oidKey); firmwareVersion != "" {
-		searchForm["FirmwareVersion"] = firmwareVersion
+	if !strings.Contains(oidKey, "/Storage") {
+		if firmwareVersion := getFirmwareVersiomn(oidKey); firmwareVersion != "" {
```
Comment (irappa-pattar) [inline svc-aggregation/system/common.go]: change the function name `getFirmwareVersiomn` to `getFirmwareVersion`

**Review by irappa-pattar — CHANGES_REQUESTED** [review]
update the function name

File: svc-aggregation/system/common.go (hunk 1 above)
Comment (amar-shalgar) [reply svc-aggregation/system/common.go]: Corrected the func name

**Review by irappa-pattar — APPROVED** [review]

**Review by bharath-b-hpe — APPROVED** [review]

File: svc-aggregation/system/rediscoverinventory.go [hunk 2]
```diff
@@ -132,11 +135,16 @@ func (e *ExternalInterface) RediscoverSystemInventory(deviceUUID, systemURL stri
 	h.TraversedLinks = make(map[string]bool)
 	progress := int32(100)
 	systemsEstimatedWork := int32(75)
-	_, progress, _ = h.getSystemInfo("", progress, systemsEstimatedWork, req)
-	//rediscovering the Chassis Information
-	req.OID = "/redfish/v1/Chassis"
-	chassisEstimatedWork := int32(15)
-	progress = h.getAllRootInfo("", progress, chassisEstimatedWork, req)
+	if strings.Contains(systemURL, "/Storage") {
+		_, progress, _ = h.getStorageInfo("", progress, systemsEstimatedWork, req)
```
Comment (shivacharanms) [inline svc-aggregation/system/rediscoverinventory.go]: If task is being used in your function why send an empty string? define your function without task

File: svc-aggregation/system/common.go [hunk 3]
```diff
@@ -643,8 +643,99 @@ func (h *respHolder) getSystemInfo(taskID string, progress int32, alottedWork in
 	return oidKey, progress, nil
 }
 
-func createServerSearchIndex(computeSystem map[string]interface{}, oidKey, deviceUUID string) map[string]interface{} {
+// getStorageInfo is used to rediscover storage data from a system
+func (h *respHolder) getStorageInfo(taskID string, progress int32, alottedWork int32, req getResourceRequest) (string, int32, error) {
```
Comment (shivacharanms) [inline svc-aggregation/system/common.go]: remove task if you are not using it

**Review by shivacharanms — CHANGES_REQUESTED** [review]

File: svc-aggregation/system/common.go (hunk 3 above)
Comment (amar-shalgar) [reply svc-aggregation/system/common.go]: Incorporated review comments

File: svc-aggregation/system/rediscoverinventory.go (hunk 2 above)
Comment (amar-shalgar) [reply svc-aggregation/system/rediscoverinventory.go]: Incorporated review comments

**Review by shivacharanms — APPROVED** [review]

**Pull request merged**